src/com/amazon/ion/impl/IonReaderContinuableTopLevelBinary.java

```diff
@@ -20,11 +21,42 @@
 import java.util.Date;
 
 /**
- * An {@link IonReader} adapter to IonReaderContinuableApplicationBinary. Currently non-continuable. Support for
- * continuability will be added.
+ * An optionally continuable (i.e., incremental) binary {@link IonReader} implementation. Continuability is enabled
+ * using {@code IonReaderBuilder.withIncrementalReadingEnabled(true)}.
+ * <p>
+ * When continuable reading is enabled, if
+ * {@link IonReader#next()} returns {@code null} at the top-level, it indicates that there is not (yet) enough data in
+ * the stream to complete a top-level value. The user may wait for more data to become available in the stream and
+ * call {@link IonReader#next()} again to continue reading. Unlike the non-incremental reader, the continuable reader
+ * will never throw an exception due to unexpected EOF during {@code next()}. If, however, {@link IonReader#close()} is
+ * called when an incomplete value is buffered, an {@link IonException} will be raised.
+ * </p>
+ * <p>
+ * There is one caveat with the continuable reader implementation: it must be able to buffer an entire top-level value
+ * and any preceding system values (Ion version marker(s) and symbol table(s)) in memory. This means that each value
+ * and preceding system values must be no larger than any of the following:
+ * <ul>
+ * <li>The configured maximum buffer size of the {@link IonBufferConfiguration}.</li>
+ * <li>The memory available to the JVM.</li>
```

Comment:
```suggestion
 * <li>The available heap memory for the JVM.</li>
```
I think the "heap" modifier is clearer, I flipped the wording because I found it awkward when I just added "heap" in front of "memory." You may prefer a different wording.

Reply:
Done.